### ISSUE TITLE
IREE model exported by JAX does not recognize static arguments as inputs

### ISSUE THREAD

**Issue by sergiuiacob1** [iree-org/iree#10671]
### What happened?

I have a compilation pipeline that works with the following versions:
```
jaxlib==0.1.61
https://github.com/google/jax/archive/refs/tags/jax-v0.2.9.tar.gz
https://github.com/iree-org/iree/releases/expanded_assets/snapshot-20210728.428
```

Pipeline is: build model with JAX -> export it for IREE -> load it using IREE runtime -> run it in python & go using the IREE runtime APIs

I am trying to upgrade IREE to latest stable (versions in "Version information" section). My issue is that static arguments are not working. I have updated the IREE interface and I am compiling as follows:

```
def aot(function, static_argnums, *args, **options):
    """Traces and compiles a function, flattening the input args.

    This is intended to be a lower-level interface for compiling a JAX function to
    IREE without setting up the runtime bindings to use it within Python. A common
    usecase for this is compiling to Android (and similar targets).

    Args:
      function: The function to compile.
      args: The inputs to trace and compile the function for.
      **kwargs: Keyword args corresponding to xla.ImportOptions or CompilerOptions
    """
    xla_comp = jax.xla_computation(function, static_argnums=static_argnums)(*args)
    hlo_proto = xla_comp.as_serialized_hlo_module_proto()
    return iree.compiler.tools.xla.compile_str(hlo_proto, input_type=iree.compiler.InputType.XLA, **options)
```

```
    trace_args = [start_step, end_step, *placeholder_inputs]
    return iree_jax_support.aot(jax_pipeline, (0, 1), *trace_args, target_backends=["llvm-cpu"], extra_args=extra_args)
```

`jax_pipeline` is built by this function:
```
def dynamic_jax_pipeline(steps, input_indexes, output_indexes):
    """
    Takes as input `steps`, which is a list of functions.
    `input_indexes` and `output_indexes` tells each step which data to work with and where to put the data

    Returns a function which takes in multiple inputs `tensors` and pipes them through each step, transforming the tensors.
    `tensors` is a variable number of 2D tensors.

    The function may also take a `start_step` and an `end_step` by which you control which sub-sequence of steps to run.
    """

    def __inner__(start_step, end_step, *tensors):
        start_step = 0 if start_step is None else start_step
        end_step = len(steps) if end_step is None else end_step

        for i, step in enumerate(steps[start_step:end_step]):
            input_data = [tensors[j] for j in input_indexes[i + start_step]]
            tensor_out = step(*input_data)
            idx = output_indexes[i + start_step]
            tensors = (*tensors[:idx], tensor_out, *tensors[idx + 1 :])
        return tensors

    return __inner__
```
where `steps` is a list of `jax.numpy` functions.

Then l am loading & running the model as follows:
```
def load_iree_binary(pipeline_bytes: bytes):
    """Loads a compiled pipeline using IREE"""
    global _DYLIB_CONFIG, _VM_INSTANCE
    if not _DYLIB_CONFIG:
        # lazy initialization – ensures the config is created in the context of the calling process
        _DYLIB_CONFIG = ireert.Config("local-task")
    if not _VM_INSTANCE:
        _VM_INSTANCE = ireert.VmInstance()

    vm_module = ireert.VmModule.from_flatbuffer(_VM_INSTANCE, pipeline_bytes)
    module_object = ireert.load_vm_module(vm_module, _DYLIB_CONFIG)
    return module_object["main"]  # the compiled function to be ran
```

```
    func1 = load_iree_binary(compiled_tensor_func[0])
    data32 = func1(data32, start_step=0, end_step=1)
```

This was working find before, now it's complaining that
```
    def __call__(self, *args, **kwargs):
      invoke_context = InvokeContext(self._device)
>     arg_list = self._arg_packer.pack(invoke_context, args, kwargs)
E     ValueError: specified kwarg 'start_step' is unknown
```

If I just put the values of the static arguments at the beginning, i.e.
```
    func1 = load_iree_binary(compiled_tensor_func[0])
    data32 = func1(0, 1, data32)
```

The error is 
```
    def __call__(self, *args, **kwargs):
      invoke_context = InvokeContext(self._device)
>     arg_list = self._arg_packer.pack(invoke_context, args, kwargs)
E     ValueError: mismatched call arity: expected 1 got 3
```

Basically I am trying to provide 2 integers which are constants at compilation time. I'm also a bit confused – should I provide these at all when running the IREE models? Or is it enough if I just provide the values for the traced inputs, the ones which are not static?

### What component(s) does this issue relate to?

Frontends, Python

### Version information

jax==0.3.17
jaxlib==0.3.15
-f https://github.com/iree-org/iree/releases/expanded_assets/candidate-20220902.254
--extra-index-url https://github.com/iree-org/iree/releases/expanded_assets/candidate-20220902.254
iree-compiler
iree-runtime
iree-tools-tf
iree-tools-tflite
iree-tools-xla

**Comment by MaheshRavishankar** [comment]
@jpienaar moving to you for initial triage.

**Comment by sergiuiacob1** [comment]
This is a duplicate of https://github.com/iree-org/iree/issues/10674
I'm sorry, I must have created the same issue 2 times.

It is sufficient if you can confirm on the other issue that my last comments make sense. Thank you!
